src/Aspire.Hosting/Dashboard/DashboardServiceHost.cs

```diff
@@ -0,0 +1,114 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Dcp;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+
+namespace Aspire.Hosting.Dashboard;
+
+/// <summary>
+/// Hosts a gRPC service (via <see cref="DashboardService"/>) that a dashboard can connect to.
+/// Configures DI and networking options for the service.
+/// </summary>
+internal sealed class DashboardServiceHost : IHostedService
+{
+    private const string DashboardServiceUrlVariableName = "DOTNET_DASHBOARD_GRPC_ENDPOINT_URL";
+    private const string DashboardServiceUrlDefaultValue = "http://localhost:18999";
+
+    /// <summary>
+    /// <see langword="null"/> if <see cref="DistributedApplicationOptions.DashboardEnabled"/> is <see langword="false"/>.
+    /// </summary>
+    private readonly WebApplication? _app;
+
+    public DashboardServiceHost(
+        DistributedApplicationOptions options,
+        DistributedApplicationModel applicationModel,
+        KubernetesService kubernetesService)
+    {
+        if (!options.DashboardEnabled)
+        {
+            return;
+        }
+
+        var builder = WebApplication.CreateBuilder();
```

Comment:
What about configuring logging for the web app?

I think the app should use the same logging as its container. I'm not sure the best way to do that off the top of my head. Maybe pass a logger factory to the DashboardServiceHost constructor and register it as a singleton?

Reply:
How was this resolved?

Reply:
Today the dashboard continues to use the app host's logging. Configuring the dashboard's logging can be done when we pull it out into a standalone tool. @mitchdenny and I have been looking into that for a future PR.